dissect/target/plugins/os/unix/_os.py

```diff
@@ -38,24 +39,102 @@ def create(cls, target: Target, sysvol: Filesystem) -> UnixPlugin:
 
     @export(record=UnixUserRecord)
     def users(self) -> Iterator[UnixUserRecord]:
-        passwd = self.target.fs.path("/etc/passwd")
-        if not passwd.exists():
-            return
+        """Recover users from /etc/passwd, /etc/master.passwd or /var/log/syslog session logins."""
 
-        for line in passwd.open("rt"):
+        # Yield users found in passwd files.
+        PASSWD_FILES = ["/etc/passwd", "/etc/master.passwd"]
+
+        for passwd_file in PASSWD_FILES:
+            if (path := self.target.fs.path(passwd_file)).exists():
+                for line in path.open("rt"):
             line = line.strip()
             if not line or line.startswith("#"):
                 continue
 
-            pwent = line.split(":")
+                    pwent = dict(enumerate(line.split(":")))
             yield UnixUserRecord(
-                name=pwent[0],
-                passwd=pwent[1],
-                uid=pwent[2],
-                gid=pwent[3],
-                gecos=pwent[4],
-                home=pwent[5],
-                shell=pwent[6],
+                        name=pwent.get(0),
+                        passwd=pwent.get(1),
+                        uid=pwent.get(2),
+                        gid=pwent.get(3),
+                        home=pwent.get(5),
+                        shell=pwent.get(6),
+                        source=passwd_file,
+                        _target=self.target,
+                    )
+
+        # Find users not in passwd files by parsing recent
+        # syslog ldap, kerberos and x-session logins.
+        if (path := self.target.fs.path("/var/log/syslog")).exists():
```

Comment:
I'm not 100% sure I'd like this to be default behaviour, how much does this impact performance on a decently sized `syslog` file? I'm leaning to this being either a separate function or behind a flag argument.

Reply:
Added feature flag `sessions` in ae86ba339c36de7d8b8ab18d8e2fa22cce6467fb.